app/jobs/geocode_venues_job.rb

```diff
@@ -0,0 +1,17 @@
+class GeocodeVenuesJob < ActiveJob::Base
+  queue_as :default
+
+  def perform
+    Venue.where(latitude: nil, longitude: nil).each do |venue|
+      results = nil
+      if venue.street_address1 && venue.city && venue.state
+        address = [venue.street_address1, venue.city, venue.state].join(',')
```

Comment:
Then, with at least one address field and then `city && state || postal_code`, we should pass everything to the geocoder (the `nil` fields will be consumed by `join`). 
`address = [venue.street_address1, venue.street_address2, venue.city, venue.state, venue.postal_code].join(',')`